server/lib/gcp.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import urllib.request
+
+url = "http://metadata.google.internal/computeMetadata/v1/project/project-id"
+
+
+def in_google_network():
+  try:
+    req = urllib.request.Request(url, headers={"Metadata-Flavor": "Google"})
+    resp = urllib.request.urlopen(req)
+    resp.read().decode()
+    return True
```

Comment:
Can we cache this in-memory so the request is made only once per session?

Reply:
I think this function as util should not cache but the caller should. That said, it is only invoked once in create_app() function during server start, so should be okay.

Reply:
Currently, yes. It could be called from multiple places in the future. Perhaps add a comment to cache if that happens.

Reply:
Done